StockServiceFactory/src/test/java/edu/mbrooks/advancedjava/stockfactory/BasicStockServiceTest.java

```diff
@@ -76,5 +75,183 @@ public void testAddQuoteNegative() {
 
     }
 
+    @Test
+    public void testQuoteHistoryHourly() {
+
+        Calendar cal = Calendar.getInstance();
+        List<StockQuote> histList = new ArrayList<>();
+        SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm");
+        StockQuote bQuote = new StockQuote("IBM", cal, 100F, 98F, "International Business Machines", "AMEX");
+        StockFactory sFactory = new StockFactory();
+        StockService IBM = sFactory.getStockService(bQuote);
+
+        try {
+            sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm");
+            cal = Calendar.getInstance();
+            cal.setTime(sdf.parse("2018-05-01 11:34"));// all done
+            IBM.addQuote(new StockQuote("IBM", cal, 100F, 98.9F, "International Business Machines", "AMEX"));
+
+            sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm");
+            cal = Calendar.getInstance();
+            cal.setTime(sdf.parse("2018-05-02 12:00"));// all done
+            IBM.addQuote(new StockQuote("IBM", cal, 101F, 100.1F, "International Business Machines", "AMEX"));
+
+            sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm");
+            cal = Calendar.getInstance();
+            cal.setTime(sdf.parse("2018-05-03 1:00"));// all done
+            IBM.addQuote(new StockQuote("IBM", cal, 99F, 98.6F, "International Business Machines", "AMEX"));
+
+            sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm");
+            cal = Calendar.getInstance();
+            cal.setTime(sdf.parse("2018-05-05 1:34"));// all done
+            IBM.addQuote(new StockQuote("IBM", cal, 104F, 103.75F, "International Business Machines", "AMEX"));
+
+            sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm");
+            cal = Calendar.getInstance();
+            cal.setTime(sdf.parse("2018-05-05 2:00"));// all done
+            IBM.addQuote(new StockQuote("IBM", cal, 103F, 102.5F, "International Business Machines", "AMEX"));
+
+            sdf = new SimpleDateFormat("yyyy-MM-dd HH:mm");
+            cal = Calendar.getInstance();
+            cal.setTime(sdf.parse("2018-05-05 2:10"));// all done
+            IBM.addQuote(new StockQuote("IBM", cal, 103.20F, 102.98F, "International Business Machines", "AMEX"));
+        } catch (Exception e) {
+            System.out.println(e);
```

Comment:
this is incorrect exception handling. 
code should never just catch and exception and log. 
if an exception happened in this code, the test should fail.

Reply:
Got it. This should've exited hard